onchain/solidity_contracts/test/ABTCVaultTest.t.sol

```diff
@@ -0,0 +1,205 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import "forge-std/Test.sol";
+import "../src/defi/ABTCVault.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract MockWrappedBTC is ERC20 {
+    constructor(string memory name, string memory symbol) ERC20(name, symbol) {
+        _mint(msg.sender, 1000000 * 10**18);
+    }
+}
+
+contract ABTCVaultTest is Test {
+    ABTCVault public vault;
+    MockWrappedBTC public wbtc;
+    MockWrappedBTC public tbtc;
+    address public admin;
+    address public user1;
+    address public user2;
+
+    function setUp() public {
+        admin = address(this);
+        user1 = address(0x1);
+        user2 = address(0x2);
+
+        wbtc = new MockWrappedBTC("Wrapped BTC", "WBTC");
+        tbtc = new MockWrappedBTC("tBTC", "TBTC");
+
+        vault = new ABTCVault();
+        vault.initialize(admin, address(wbtc));
+
+        vm.label(address(vault), "ABTCVault");
+        vm.label(address(wbtc), "WBTC");
+        vm.label(address(tbtc), "TBTC");
+        vm.label(user1, "User1");
+        vm.label(user2, "User2");
+    }
+
+    function testInitialization() public {
+        assertEq(vault.name(), "Aggregated Bitcoin");
+        assertEq(vault.symbol(), "aBTC");
+        assertEq(vault.decimals(), 18);
+        assertTrue(vault.hasRole(vault.DEFAULT_ADMIN_ROLE(), admin));
+        assertTrue(vault.hasRole(vault.MINTER_ROLE(), admin));
+        assertTrue(vault.hasRole(vault.UPGRADER_ROLE(), admin));
+        assertTrue(vault.hasRole(vault.PAUSER_ROLE(), admin));
+    }
+
+    function testSetWrappedBTCToken() public {
+        vault.setWrappedBTCToken(address(wbtc), true, 1e18, block.timestamp);
+        (bool isPermitted, uint256 ratio, uint256 poolingTimestamp) = vault.wrappedBTCTokens(address(wbtc));
+        assertTrue(isPermitted);
+        assertEq(ratio, 1e18);
```

Comment:
Can we change the ratio as a fee percentage?
Setup to 0 first and we can change it after